src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Semantic/SemanticTokenTestBase.cs

```diff
@@ -105,33 +103,34 @@ protected string GetBaselineFileContents(string baselineFileName)
         await using var csharpServer = await CSharpTestLspServerHelpers.CreateCSharpLspServerAsync(
             csharpSourceText, csharpDocumentUri, SemanticTokensServerCapabilities, SpanMappingService, DisposalToken);
 
-        var csharpRanges = GetMappedCSharpRanges(codeDocument, razorRange, out var minimalRange);
-        if (minimalRange == null)
+        var csharpRanges = GetMappedCSharpRanges(codeDocument, razorRange);
+        if (csharpRanges == null)
         {
-            return new ProvideSemanticTokensResponse(tokens: Array.Empty<int>(), hostDocumentSyncVersion);
+            return new ProvideSemanticTokensResponse(tokens: Array.Empty<int>(), hostDocumentSyncVersion: 0);
         }
 
-        SemanticTokens? result;
-        try
+        if (UsePreciseSemanticTokenRanges)
         {
-            result = await csharpServer.ExecuteRequestAsync<SemanticTokensRangesParams, SemanticTokens>(
-            "roslyn/semanticTokenRanges",
-            CreateVSSemanticTokensRangesParams(csharpRanges, csharpDocumentUri),
-            DisposalToken);
+            var result = await csharpServer.ExecuteRequestAsync<SemanticTokensRangesParams, SemanticTokens>(
+                "roslyn/semanticTokenRanges",
+                CreateVSSemanticTokensRangesParams(csharpRanges, csharpDocumentUri),
+                DisposalToken);
+
+            return new ProvideSemanticTokensResponse(tokens: result?.Data, hostDocumentSyncVersion: 0);
         }
-        catch (RemoteMethodNotFoundException)
```

Comment:
Since this is removed would it make sense to also remove the logic to fallback to old endpoint here?

https://github.com/dotnet/razor/blob/main/src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/RazorSemanticTokensInfoService.cs#L321

Reply:
I think it would make sense, yes, but whether or now we can do it would be a question I would ask you 😛

Weren't you looking into telemetry that seemed to indicate the fallback was being used? Reasoning about test changes is easy, because its a known environment, I didn't try to make any big product changes with this PR.

Reply:
that's fair. we noticed some failures with the feature experiment included. I am gonna check callstacks of the failures to get confirmation on whether the failures exercise the fallback logic or not